feat: Add model pinning feature to protect critical models from LRU eviction

malaiwah opened the pull request:
Adds a pin (`pin = true`) option to protect models (from the presets file) from LRU eviction in router mode. This lets some models be "always-on" and can be used at the same time as `load-on-startup = true`.

Commit:
Add pin option to protect models from LRU eviction in router mode

- Add COMMON_ARG_PRESET_PIN define
- Add pin preset option for model protection
- Add pinned field to server_model_meta
- Modify load_models to set pinned flag
- Update unload_lru to skip pinned models

ngxson commented on tools/server/server-models.cpp:
```diff
@@ -383,7 +391,7 @@ void server_models::unload_lru() {
             if (m.second.meta.is_active()) {
                 count_active++;
-                if (m.second.meta.last_used < lru_last_used) {
+                if (!m.second.meta.pinned && m.second.meta.last_used < lru_last_used) {
```
undefined behaviors:
- what if all models has `pin`
- what if all **loaded** models has `pin`

ngxson replied:
I don't think violating max-models is an acceptable behavior. Throw an error instead

Commit:
Add warning when LRU eviction cannot succeed due to all models being pinned

When models_max limit is reached but all active models are pinned, log a
warning message to clarify that automatic unload cannot succeed.

... also add --pin preset option documentation.

Commit:
Throw error when models_max limit reached with no unpinned models for LRU eviction
